includes/Context.php

```diff
@@ -197,6 +197,20 @@ public function get_reference_permalink( $post = 0 ) {
 		return $permalink;
 	}
 
+	/**
+	 * Gets the canonical url for the current request.
+	 */
+	public function get_reference_canonical() {
+		global $wp;
+		$reference_permalink = $this->get_reference_permalink();
+		if ( ! $reference_permalink ) {
+
+			// Fall back to site reference URL + current path.
+			return trailingslashit( $this->get_reference_site_url() . $wp->request );
```

Comment:
Let's use a more specific URL lookup based on conditions, the current `$wp->request` is not necessarily the canonical. As outlined in the IB, this method should return `false` for anything not (yet) covered.

For now, let's add here:
* If `is_front_page()`, return `$this->get_reference_site_url()`.
* Elseif `is_home()`, return `$this->get_reference_permalink( get_option( 'page_for_posts' ) )`.

We can later expand to more things like, taxonomy archives or post type archives, etc. All of these checks should only be made if `! is_admin()`, because these conditionals are only relevant for the frontend.

Also, because of the more complex need for these conditions, let's reverse the logic a bit to have less nesting, e.g.
```php
if ( $reference_permalink || is_admin() ) {
    return $reference_permalink;
}
```

Reply:
👍 	Will do, I wasn't sure about using `$wp->request` here.

Reply:
Handled in https://github.com/google/site-kit-wp/pull/543/commits/b53eb79f141f17bd1612bf01e0975c65170db1e3